src/main/scala/com/fulcrumgenomics/fasta/CollectAlternateContigNames.scala

```diff
@@ -0,0 +1,175 @@
+  @arg(flag='s', doc="Only output sequences with the given sequence roles.  If none given, all sequences will be output.", minElements=0)
+  val sequenceRoles: Seq[SequenceRole] = Seq.empty,
+  @arg(doc="Output all alternates on a single line.") val singleLine: Boolean = false
+) extends FgBioTool with LazyLogging {
+
+  Io.assertReadable(input)
+  Io.assertCanWriteFile(output)
+  validate(!alternates.contains(primary), s"Primary is in alternate: $primary in " + alternates.mkString(", "))
+
+  override def execute(): Unit = {
+    // Go through each sequence in the assembly report
+    val lines = readAssemblyReport().toIterator.flatMap { dict =>
+      // Get the primary name, and the alternate names
+      val primary = dict(this.primary.key)
+      val altNames = this.alternates.flatMap { alt =>
+        dict(alt.key) match {
+          case alternate if alternate == AssemblyReportColumn.MissingValue =>
+            logger.warning(s"Missing value for alternate ${alt.key}: $primary")
+            None
+          case alternate => Some(alternate)
+        }
+      }
+      (primary, altNames) match {
+        case (AssemblyReportColumn.MissingValue, _) => // primary missing
+          logger.warning(s"Primary contig name '${this.primary.key}' had a missing value: $primary")
+          Seq.empty
+        case (_, Seq()) => // no alternates
+          logger.warning(s"No alternates found for: $primary")
+          Seq.empty
+        case _ => // primary and at least one alternate
+          if (singleLine) {
+            Seq(primary + "\t" + altNames.mkString("\t"))
+          }
+          else {
+            altNames.map { altName => s"$primary\t$altName" }
+          }
+      }
+    }
+
+    Io.writeLines(output, lines)
+  }
+
+  private def readAssemblyReport(): Seq[Map[String, String]] = {
+    val iter = Io.readLines(input).bufferBetter
+
+    // skip over comments until we reach the header
+    iter.dropWhile { line => line.startsWith("#") && !line.startsWith(s"# ${AssemblyReportColumn.SequenceName.key}") }
+
+    // read the header
+    require(iter.hasNext, s"Missing header from $input.")
+    val header = iter.next().split('\t')
+
+    // Reads the input assembly report into a sequence of key/value maps
+    val maps = iter.flatMap { line =>
+      val fields = line.split('\t')
+      val d      = header.zip(fields).toMap
+      if (sequenceRoles.isEmpty || sequenceRoles.exists(_.key == d(AssemblyReportColumn.SequenceRole))) {
+        Some(d)
+      } else {
+        None
+      }
+    }.toSeq
+
+    // Re-order the entries based on primary vs non primary sequence roles, then based on increasing sequence length
+    val primary   = maps.filter { d => SequenceRole(d(AssemblyReportColumn.SequenceRole)).primary }
+    val secondary = maps.filter { d => !SequenceRole(d(AssemblyReportColumn.SequenceRole)).primary}.sortBy { d => d(AssemblyReportColumn.SequenceLength)}
```

Comment:
Will this sort be lexographic if the AssemblyReportColumn.SequenceLength is a string?

Reply:
Yes it would be, I'm going to convert it to an Int. @nh13, does that sound correct to you?

Reply:
After looking at the tests, this does seem to actually want to be in lexicographical order